mentat/code_changes/abstract/abstract_change.py

```diff
@@ -0,0 +1,225 @@
+import logging
+from functools import cmp_to_key
+from pathlib import Path
+from typing import cast
+
+import attr
+from termcolor import cprint
+
+from mentat.code_context import CodeContext
+from mentat.code_file import CodeFile
+from mentat.errors import MentatError
+from mentat.user_input_manager import UserInputManager
+
+
+@attr.s
+class Addition:
+    # Will insert directly before this line, 0 indexed
+    line_number: int = attr.field()
+    content: list[str] = attr.field()
+    owner: int = attr.field()
+
+
+@attr.s
+class Deletion:
+    # Inclusive, 0 indexed
+    starting_line: int = attr.field()
+    # Exclusive, 0 indexed
+    ending_line: int = attr.field()
+    owner: int = attr.field()
+
+
+@attr.s
+class Rename:
+    # None represents deleting this file
+    name: Path | None = attr.field()
+    owner: int = attr.field()
```

Comment:
I don't love using `None` for deletion inside `Rename`. I might prefer explicit `DeleteFile`, `CreateFile`, `RenameFile` classes. Or maybe the name could be `FileUpdate` or something like that instead of `Rename`

Reply:
Hmm, looking back at it I think I agree that having DeleteFile CreateFile and RenameFile classes would be nicer and simpler; I'll switch over to that.

Reply:
Hmm, but thinking about it more I do like having just one class